packages/ucp-enforcement/src/UcpPatternEnforcement.ts

```diff
@@ -0,0 +1,132 @@
+        // console.log("input:");
+        // console.log(await rdfTransformStore(reasoningInputStore, 'text/turtle'));
+
+        // Reason using ODRL Rules and N3 Rules
+        const reasoningResult = await this.reasoner.reason(reasoningInputStore, this.koreografeyeOdrlRules);
+
+        // TODO: remove in production
+        // console.log("reasoning output:");
+        // console.log(await rdfTransformStore(reasoningResult, 'text/turtle'));
+
+        // Execute policies
+        const accessModes: AccessMode[] = []
+        const executedPolicies = await this.executor.executePolicies(reasoningResult)
+        // if no policies -> ask owner for request -> plugin?
+        if (executedPolicies.length === 0) {
+            // no access
+            // TODO: ask owner access
+            // TODO: let user know that there is no access
+            console.log(`[${new Date().toISOString()}] - UcpPatternEnforcement: no policies`);
+        }
+        // if policies -> executePolicy: return value accessmodes in an object somehow?
+        for (const executedPolicy of executedPolicies) {
+            accessModes.push(...executedPolicy.result)
+        }
+        return accessModes
+    }
+
+    async calculateAndExplainAccessModes(request: UconRequest): Promise<Explanation> {
+        const contextStore = createContext(request)
+
+        const knowledgeBase = new Store()
+
+        knowledgeBase.addQuads((await this.uconRulesStorage.getStore()).getQuads(null, null, null, null))
+        knowledgeBase.addQuads(contextStore.getQuads(null, null, null, null))
+
+        const reasoningResult = await this.reasoner.reason(knowledgeBase, this.koreografeyeOdrlRules);
+
+        const conclusions: Conclusion[] = []
+        const executedPolicies = await this.executor.executePolicies(reasoningResult)
+
+        if (executedPolicies.length === 0) {
+            // based on all the rules, no access
+        }
+
+        for (const executedPolicy of executedPolicies) {
+            // expect that the result of the executedPolicy consists of:
+            //  * usage control rule identifier
+            //  * N3 interpretation rule identifier
+            //  * time issued
+            //  * access modes granted
+
+            // if not, throw Error -> the wrong plugin is used | Note: throwing error is not happening yet
+
+            conclusions.push(executedPolicy.result)
+        }
+
+        // calculation of decision based on the algorithm and the conclusions
+        // Note: currently hardcoded to be the union.
+        const grants = conclusions.map(conclusion => conclusion.grants).flat()
+        // remove the duplicates
+        const decision = Array.from(new Set(grants))
+        return {
+            decision: decision,
+            request: {
+                raw: request
+            },
+            algorithm: DecisionAlgorithm.Union,
+            conclusions: conclusions,
+        }
+    }
+}
+
+export interface UconEnforcementDecision {
```

Comment:
Do we really need an interface here? It seems very specific. If we really want to extend it, better put it in a separate file.